### ISSUE TITLE
Error when resampling forcing in time

### ISSUE THREAD

**Issue by hboisgon** [Deltares/hydromt#1035]
### HydroMT version checks

- [X] I have checked that this issue has not already been reported.
- [X] I have checked that this bug exists on the latest version of HydroMT.

### Reproducible Example

Try using precip or temp workflow with 1h era5 data and prepare data for 3H timestep.
Eg from hydromt-wflow that is using this with deltares_data catalog:

```yaml
setup_config:                 # options parsed to wflow toml file <section>.<option>
  starttime: 2020-01-01T00:00:00
  endtime: 2020-02-01T00:00:00
  timestepsecs: 86400
  input.path_forcing: inmaps-era5-daily_from_hourly.nc
 
setup_precip_forcing:
  precip_fn: era5_hourly          # source for precipitation.
  precip_clim_fn:          # source for high resolution climatology to correct precipitation if any.

setup_temp_pet_forcing:
  temp_pet_fn: era5_hourly        # source for temperature and potential evapotranspiration.
  press_correction: True          # if True temperature is corrected with elevation lapse rate.
  temp_correction: True          # if True pressure is corrected with elevation lapse rate.
  dem_forcing_fn: era5_orography # source of elevation grid corresponding to temp_pet_fn. Used for lapse rate correction.
  pet_method: debruin       # method to compute PET: {debruin, makkink}
  skip_pet: False         # if True, only temperature is prepared.
```

### Current behaviour

Log:
```
2024-08-23 11:44:03,882 - update - forcing - DEBUG - downsampling precip using sum; conserve mass: True
2024-08-23 11:44:03,883 - update - main - ERROR - freq must be a str or a Resampler object
Traceback (most recent call last):
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\hydromt\cli\main.py", line 328, in update
    mod.update(model_out=model_out, opt=opt, forceful_overwrite=fo)
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\hydromt\models\model_api.py", line 329, in update
    self._run_log_method(method, **kwargs)
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\hydromt\models\model_api.py", line 188, in _run_log_method
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\hydromt_wflow\wflow.py", line 2400, in setup_precip_forcing
    precip_out = hydromt.workflows.forcing.precip(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\hydromt\workflows\forcing.py", line 85, in precip
    p_out = resample_time(p_out, freq, conserve_mass=True, **resample_kwargs)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\hydromt\workflows\forcing.py", line 751, in resample_time
    da_resampled = da.resample(time=freq, skipna=True, label=label, closed=closed)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\xarray\util\deprecation_helpers.py", line 115, in inner
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\xarray\core\dataarray.py", line 7384, in resample
    return self._resample(
           ^^^^^^^^^^^^^^^
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\xarray\core\common.py", line 1074, in _resample
    raise ValueError("freq must be a str or a Resampler object")
ValueError: freq must be a str or a Resampler object
Traceback (most recent call last):
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Scripts\hydromt-script.py", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\click\core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\click\core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\click\core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\click\core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\click\core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\click\decorators.py", line 33, in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\hydromt\cli\main.py", line 328, in update
    mod.update(model_out=model_out, opt=opt, forceful_overwrite=fo)
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\hydromt\models\model_api.py", line 329, in update
    self._run_log_method(method, **kwargs)
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\hydromt\models\model_api.py", line 188, in _run_log_method
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\hydromt_wflow\wflow.py", line 2400, in setup_precip_forcing
    precip_out = hydromt.workflows.forcing.precip(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\hydromt\workflows\forcing.py", line 85, in precip
    p_out = resample_time(p_out, freq, conserve_mass=True, **resample_kwargs)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\hydromt\workflows\forcing.py", line 751, in resample_time
    da_resampled = da.resample(time=freq, skipna=True, label=label, closed=closed)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\xarray\util\deprecation_helpers.py", line 115, in inner
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\xarray\core\dataarray.py", line 7384, in resample
    return self._resample(
           ^^^^^^^^^^^^^^^
  File "C:\Users\boisgont\Miniconda3\envs\hydromt-test\Lib\site-packages\xarray\core\common.py", line 1074, in _resample
    raise ValueError("freq must be a str or a Resampler object")
ValueError: freq must be a str or a Resampler object
```

### Desired behaviour

Seems to be due to the new xarray version v2024.07.0: https://docs.xarray.dev/en/stable/whats-new.html#v2024-07-0-jul-30-2024

Downgrading xarray solves the issue

**Comment by hboisgon** [comment]
Another related issue with the new xarray version happens in raster.interpolate_na. See https://github.com/Deltares/hydromt_wflow/issues/287

**Comment by hboisgon** [comment]
Seems to have been fixed in the new release of xarray 2024.09
